projects/hangman/hangman_functions.py

```diff
@@ -0,0 +1,219 @@
+       :param list used_letters : This list contains all alphabets that are already used by user
+
+       :returns:  True or False depending on that the guessed letter is right or wrong respectively
+       :return type: Boolean
+    """
+    check = False
+
+    for letters in user_letter:
+        if letters.islower():
+            user_letter = user_letter.upper()
+        else:
+            pass
+
+    print('                                                    ', end=' ')
+
+    for i in range(len(guess_list)):
+        if user_letter == guess_list[i]:
+            memory_list[i] = user_letter
+            check = 'True'
+        else:
+            pass
+
+    for letters in memory_list:
+        print(letters, end=' ')
+
+    if check:
+        print('             YESS; YOU GOT RIGHT', end=' ')
+    else:
+        print('             OOPPSSS; YOU MISSED', end=' ')
+
+    print('                 USED LETTERS:', end=' ')
+    for letters in used_letters:
+        if letters.isupper():
+            print(letters, end=' ')
+
+    print('\n')
+
+    return check
+
+
+def greeting_function():
+    """
+       Function for printing a greeting message and instructions to play the game
+
+       :param : none
+
+       :returns:  None
+       :return type: None
+    """
+    flag = True
+    print('\n')
+    print('                                                       *** WELCOME TO HANGMAN GAME ***')
+    print('\n')
+    print('                                              <><><> LETS UNDERSTAND HOW TO PLAY THE GAME <><><>')
+    print('\n')
+    print('This is a type of word guessing game. First you have the choice to choose\n'
+          'from four categories. After choosing your favourite one, a blank word will\n'
+          'appear on your screen to guess. You will have limited turns to guess the word.\n'
+          'If within allowed turns, you guess the word correctly, YOU WIN :-)\n'
+          'Otherwise you will be HANGED :-(\n')
+
+    while flag:
+        understood = input('PLEASE PRESS LETTER (P) IF YOU WANT TO START THE GAME!')
+        if understood in ['P', 'p']:
+            break
+        else:
+            print('SORRY! YOU HAVE ENTERED WRONG LETTER. TRY AGAIN')
+
+
+def display_categories():
+    """
+       Function for displaying four categories to choose from
+
+       :param : None
+
+       :returns:  None
+       :return type: None
+    """
+    print('\n')
+    print('YOU HAVE FOLLOWING FOUR CATEGORIES TO CHOOSE FROM:')
+    print('COUNTRIES    (Press 1)\nSCIENCE      (Press 2)\nFRUITS       (Press 3)\nPROFESSIONS  (Press 4)')
+
+
+def user_input():
+    """
+       Function for asking user to select his favourite category out of four
+
+       :param : None
+
+       :returns:  Selected category of user
+       :return type: String
+    """
+    flag = True
+    selected_category = str()
+
+    while flag:
+        user_category_input = input('PLEASE SELECT YOUR CATEGORY')
```

Comment:
also for better user experience you can give a space or `:/;` when you are expecting a user input

![image](https://user-images.githubusercontent.com/12052941/104838188-906a1f00-58b9-11eb-8246-92133a8435e9.png)

Reply:
Done

Reply:
Done